Autogen headerfile gaurds break cspec

### Steps 
Editing a new headerfile, kakoune auto-generates headerguards using the filename, eg: `kak headerfile.h` produces:
```
#ifndef headerfile_h_INCLUDED
#define headerfile_h_INCLUDED

#endif headerfile_h_INCLUDED
```
(assuming `headerfile.h` doesn't already exist)
This is great and very helpful, but if you have path seperator in the name, eg: `kak src/newheader.h`, the header guards become:
```
#ifndef src/newheader_h_INCLUDED
#define src/newheader_h_INCLUDED

#endif src/newheader_h_INCLUDED
```

The c compiler does not like this. This actually internally defines `src`, not `src/newheader_h_INCLUDED`, meaning the header guards won't work properly if there is another headerfile in the same directory. An example gcc ouput:
```
~/tmp❱ gcc test.c
In file included from test.c:1:
subdir/header.h:1:15: warning: extra tokens at end of #ifndef directive
    1 | #ifndef subdir/header_h_INCLUDED
      |               ^
subdir/header.h:2:9: warning: ISO C99 requires whitespace after the macro name
    2 | #define subdir/header_h_INCLUDED
      |         ^~~~~~
```

### Expected
Headerguards should probably just use the filename, and ignore the path around them

--- a/rc/filetype/c-family.kak
+++ b/rc/filetype/c-family.kak
@@ -387,7 +387,7 @@ define-command -hidden c-family-insert-include-guards %{
     evaluate-commands %sh{
         case "${kak_opt_c_include_guard_style}" in
             ifdef)
-                echo 'execute-keys ggi<c-r>%<ret><esc>ggxs\.<ret>c_<esc><space>A_INCLUDED<esc>ggxyppI#ifndef<space><esc>jI#define<space><esc>jI#endif<space>//<space><esc>O<esc>'
+                echo 'execute-keys ggi<c-r>%<ret><esc>ggI/<esc>xs^.*/<ret>dxs\.<ret>c_<esc><space>A_INCLUDED<esc>ggxyppI#ifndef<space><esc>jI#define<space><esc>jI#endif<space>//<space><esc>O<esc>'
                 ;;
             pragma)
                 echo 'execute-keys ggi#pragma<space>once<esc>'

